PWGHF/HFC/TableProducer/correlatorDMesonPairs.cxx

```diff
@@ -34,693 +32,762 @@ using namespace o2::constants::physics;
-const double yAxisMax = 5.;
-const int ptDAxisBins = 180;
-const double ptDAxisMin = 0.;
-const double ptDAxisMax = 36.;
+enum PairTypeOfSelMassSel {
+  DD = 0,   // This is a D0-D0 pair
+  DbarDbar, // This is a D0bar-D0bar pair
+  DDbar,    // This is a D0-D0bar pair
+  DbarD     // This is a D0bar-D0 pair
+};
+} // namespace
 
 using McParticlesPlus2Prong = soa::Join<aod::McParticles, aod::HfCand2ProngMcGen>;
-using McParticlesPlus3Prong = soa::Join<aod::McParticles, aod::HfCand3ProngMcGen>;
 
 struct HfCorrelatorDMesonPairs {
   SliceCache cache;
   Preslice<aod::HfCand2Prong> perCol2Prong = aod::hf_cand::collisionId;
-  Preslice<aod::HfCand3Prong> perCol3Prong = aod::hf_cand::collisionId;
 
   Produces<aod::D0Pair> entryD0Pair;
-  Produces<aod::D0PairRecoInfo> entryD0PairRecoInfo;
-  Produces<aod::DplusPair> entryDplusPair;
-  Produces<aod::DplusPairRecoInfo> entryDplusPairRecoInfo;
+  Produces<aod::D0PairMcInfo> entryD0PairMcInfo;
+  Produces<aod::D0PairMcGen> entryD0PairMcGen;
+  Produces<aod::D0PairMcGenInfo> entryD0PairMcGenInfo;
 
-  Configurable<bool> verbose{"verbose", false, "Enable debug mode"};
-  // Selection Flags
   Configurable<int> selectionFlagD0{"selectionFlagD0", 1, "Selection Flag for D0"};
   Configurable<int> selectionFlagD0bar{"selectionFlagD0bar", 1, "Selection Flag for D0bar"};
-  Configurable<int> selectionFlagDplus{"selectionFlagDPlus", 1, "Selection Flag for DPlus"};
-  // Kinematics
-  Configurable<float> etaCandMax{"etaCandMax", 4.0, "max. eta accepted"};
-  Configurable<float> etaCandMin{"etaCandMin", -4.0, "min. eta accepted"};
-  Configurable<float> dcaXYMax{"dcaXYMax", 0.0025, "max. dcaXY accepted"};
-  Configurable<float> dcaZMax{"dcaZMax", 0.0025, "max. dcaZ accepted"};
-
-  Configurable<float> yCandMax{"yCandMax", -1., "max. cand. rapidity"};
-  Configurable<float> ptCandMin{"ptCandMin", -1., "min. cand. pT"};
-  Configurable<float> multMin{"multMin", 0., "minimum multiplicity accepted"};
-  Configurable<float> multMax{"multMax", 10000., "maximum multiplicity accepted"};
+  Configurable<float> yCandMax{"yCandMax", 0.8, "maxmum |y| of D0 candidates"};
+  Configurable<float> ptCandMin{"ptCandMin", -1., "minimum pT of D0 candidates"};
+  Configurable<std::vector<double>> binsPt{"binsPt", std::vector<double>{o2::analysis::hf_cuts_d0_to_pi_k::vecBinsPt}, "pT bin limits for candidate mass plots"};
   Configurable<bool> selectSignalRegionOnly{"selectSignalRegionOnly", false, "only use events close to PDG peak"};
   Configurable<float> massCut{"massCut", 0.05, "Maximum deviation from PDG peak allowed for signal region"};
-  Configurable<std::vector<double>> binsPt{"binsPt", std::vector<double>{o2::analysis::hf_cuts_d0_to_pi_k::vecBinsPt}, "pT bin limits for candidate mass plots"};
+  Configurable<bool> daughterTracksCutFlag{"daughterTracksCutFlag", false, "Flag to add cut on daughter tracks"};
 
   HfHelper hfHelper;
 
+  using TracksWPid = soa::Join<aod::Tracks, aod::TracksPidPiExt, aod::TracksPidKaExt>;
```

Comment:
Hi @atavirag I think that you are using the "old" tables here, that were replaced by these ones https://github.com/AliceO2Group/O2Physics/blob/2feeea18b276e0f2e6e971777b6dc3236cee36b0/PWGHF/DataModel/CandidateReconstructionTables.h#L180-L201 (see https://github.com/AliceO2Group/O2Physics/pull/5065)

This should fix the compilation error you get.

Reply:
Yes, sorry, I used an older version of O2Physics to check if it compiled